apps/snitch_core/lib/core/domain/taxonomy/taxonomy.ex

```diff
@@ -275,7 +287,47 @@ defmodule Snitch.Domain.Taxonomy do
   @doc """
   Delete a taxon
   """
-  def delete_taxon(taxon) do
-    taxon |> AsNestedSet.delete() |> AsNestedSet.execute(Repo)
+  def delete_taxon(taxon_id) do
+    case get_taxon(taxon_id) do
+      %Taxon{} = taxon ->
+        Multi.new()
+        |> Multi.run(:delete_products, fn _ -> do_delete_products(taxon_id) end)
+        |> Multi.run(:category, fn _ -> do_delete_taxon(taxon) end)
+        |> Repo.transaction()
+
+      nil ->
+        {:error, :not_found}
+    end
+  end
+
+  defp do_delete_products(taxon_id) do
+    categories =
+      get_all_children_and_self(taxon_id)
+      |> Enum.map(& &1.id)
+
+    product_delete_query =
+      from(p in Product,
+        where: p.taxon_id in ^categories
+      )
+
+    total_products = from(p in product_delete_query, select: count(p.id)) |> Repo.one()
+
+    {update_products_count, t} =
+      Repo.update_all(product_delete_query,
+        set: [state: "deleted", deleted_at: NaiveDateTime.utc_now(), taxon_id: nil]
```

Comment:
@SagarKarwande @aviacommerce/snitch-core this is a very important thing. We CANNOT have the logic of product delete in more than 1 place. The logic to set the `delete_at` to current_time. 

This is violating the `DRY` principle. Please make sure not do this type of things.